cache/cache.go

```diff
@@ -57,3 +60,24 @@ func (c *Cache) WriteTo(w io.Writer) (int64, error) {
 	}
 	return total, nil
 }
+
+// Index generates an index for the given cache on a particular
+// column.  This is required for caches beyond a libnss-cache defined
+// size in order for them to be read correctly.
+func (c *Cache) Index(col int) bytes.Buffer {
+	var b bytes.Buffer
+	offset := 0
+	for i := range c.entries {
+		rline := c.entries[i].String()
+		key := strings.Split(rline, ":")[col]
```

Comment:
I can add functions on Entry to get the specified columns, but then I'll also need a function that computes the length of the string representation of each line.  Off the top of my head I'm not aware of an idiot-proof way to do this short of rendering the string and taking the length of it.  I'll give it another look in the morning though to see what I can come up with.

Reply:
Ah yes, I missed the fact that you needed the length of the string to compute the offset.
But I think it's still best to avoid the `strings.Split()`.